lib/util.ts

```diff
@@ -49,7 +49,8 @@ export const strToChars = (str: string): string => {
  */
 export const tokenizeClass = (str: string, regexpStr: string): [SetTokens, number] => {
   let tokens: SetTokens = [], rs: string[] | null, c: string;
-  const regexp = /\\(?:(w)|(d)|(s)|(W)|(D)|(S))|((?:(?:\\)(.)|([^\]\\]))-(?:\\)?([^\]]))|(\])|(?:\\)?([^])/g;
+  const regexp =
+  /\\(?:(w)|(d)|(s)|(W)|(D)|(S))|((?:(?:\\)(.)|([^\]\\]))-(((?:\\)])|(((?:\\)?([^\]])))))|(\])|(?:\\)?([^])/g;
```

Comment:
i regret parsing classes this way >_<

just a note, not a review of your changes. at some point, i want to refactor this function to tokenize character by character, not through a regex which can look confusing